lib/comparators/hashed_configuration.rb

```diff
@@ -0,0 +1,5 @@
+module HashedConfiguration
+  def hash_configuration(opts)
```

Comment:
If you convert this mixin into a normal abstract class, and call this method within `initialize` here, your code could be much simpler. 

For example: 

```ruby
class HashedConfiguration 
    def initialize(opts)
         parse_options(hash_configuration(opts))
   end
   ...
end      

class RegexpComparator <  HashedConfiguration
 
  def parse_options(options)
        regexp = eval_regexp(options[:expected])
        raise 'No regexp found' unless regexp.is_a? Regexp
       @expected_regexp = regexp
  end
  ...
end
```